### ISSUE TITLE
AudioStreamRandomPitch but fixed pitch?

### ISSUE THREAD

**Issue by ghost** [godotengine/godot#10094]
**Operating system or device - Godot version:**
Godot 3 Feature Request

**Issue description:**

I notice that AudioStreamRandomPitch doesn't only change pitch like AudioEffectPitchShift. It also changes tempo, which might be a desirable audio effect in game. However, users have no control over this pitch/tempo value as it's random at playback time. AFAIK there's no AudioEffect that modifies tempo.

Basically, I want AudioStreamRandomPitch with user-defined pitch/tempo.

**Comment by reduz** [comment]
Doing this for AudioStreamRandomPitch would impose a very high performance limitation, as separate pitch/tempo shifting is very expensive computationally and not as precise.

What I was thinking about doing is an AudioStreamEffectStack, which is basically an audio stream that lets you apply a bunch of effects similarly to a bus.

**Comment by Noshyaar** [comment]
Does that mean there will be an AudioEffect that modifies Stream's tempo? Because changing pitch only is already possible.

**Comment by reduz** [comment]
kicking to 3.1, too many things on the plate
